### PR TITLE
Add comments, defensive error handling, and tests to frequency_encoder()

### LINKED ISSUE #53: Add more comments to all functions and tests
- [x] Conjugate encoder 

- [x] Target encoder

- [x] Frequency encoder

- [x] One hot encoder

### LINKED ISSUE #54: Add defensive code and associated tests to functions

### PR DESCRIPTION
**Pull request by bbaillie**

### CODE REVIEW

File: tests/testthat/test-frequency_encoder.R [hunk 1]
```diff
@@ -34,3 +34,17 @@ test_that("frequency_encoder() does not produce a list with correct dimensions",
...
 })
+
+test_that("frequency_encoder() does not throw an error if we input invalid arguments", {
+  testing_data <- readr::read_csv("../testdata/testing_data.csv")
+  # Test for categorical columns not being in X_train
+  expect_error(conjugate_encoder(
+    X_train = testing_data,
+    X_test = testing_data,
+    cat_columns = c("cool4cats")))
+  # check type of input X_train
```
Comment (braydentang1) [inline tests/testthat/test-frequency_encoder.R]: The check is checking against conjugate_encoder, not frequency_encoder.
```suggestion
  expect_error(frequency_encoder(
    X_train = testing_data,
    X_test = testing_data,
    cat_columns = c("cool4cats")))
  # check type of input X_train
```

**Review by braydentang1 — CHANGES_REQUESTED** [review]

File: tests/testthat/test-frequency_encoder.R (hunk 1 above)
Comment (bbaillie) [reply tests/testthat/test-frequency_encoder.R]: Oops, didn't catch that. Thank you!

File: tests/testthat/test-frequency_encoder.R [hunk 2]
```diff
@@ -34,3 +34,17 @@ test_that("frequency_encoder() does not produce a list with correct dimensions",
...
+  expect_error(frequency_encoder(
+    X_train = testing_data,
+    X_test = testing_data,
+    cat_columns = c("cool4cats")))
+  # check type of input X_train
+  expect_error(target_encoder(
+    X_train = c(1,2,3),
+    X_test = testing_data,
+    cat_columns = c("feature_cat_chr")))
+})
```
Comment (braydentang1) [inline tests/testthat/test-frequency_encoder.R]:
```suggestion
  expect_error(frequency_encoder(
    X_train = c(1,2,3),
    X_test = testing_data,
    cat_columns = c("feature_cat_chr")))
})
```

**Review by braydentang1 — CHANGES_REQUESTED** [review]

File: tests/testthat/test-frequency_encoder.R (hunk 2 above)
Comment (bbaillie) [reply tests/testthat/test-frequency_encoder.R]: One more haha, thank you

**Review by braydentang1 — APPROVED** [review]
Thanks.

**Pull request merged**